src/Compute/custom/NewAzVM_SimpleParameterSet.cs

```diff
@@ -0,0 +1,346 @@
+
+
+        [Parameter(Mandatory = false)]
+        public string Size { get; set; } = "Standard_DS1_v2";
+
+        [Parameter(Mandatory = false)]
+        public string AvailabilitySetName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Use this to add system assigned identity (MSI) to the vm")]
+        public SwitchParameter SystemAssignedIdentity { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Use this to add the assign user specified identity (MSI) to the VM")]
+        [ValidateNotNullOrEmpty]
+        public string UserAssignedIdentity { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int[] DataDiskSizeInGb { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter EnableUltraSSD { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string ProximityPlacementGroup { get; set; }
+
+        public CancellationToken Token => Source.Token;
+
+        public Action Cancel => Source.Cancel;
+
+        protected override void StopProcessing()
+        {
+            this.Cancel();
+            base.StopProcessing();
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+            this.StartAndWait(StrategyExecuteCmdletAsync);
+        }
+
+        class Parameters : IParameters<VirtualMachine>
+        {
+            readonly NewAzVM_SimpleParameterSet _cmdlet;
+
+            readonly Client _client;
+
+            readonly IResourceManagementClient _resourceClient;
+
+            public Parameters(NewAzVM_SimpleParameterSet cmdlet, Client client, IResourceManagementClient resourceClient)
+            {
+                _cmdlet = cmdlet;
+                _client = client;
+                _resourceClient = resourceClient;
+            }
+
+            public ImageAndOsType ImageAndOsType { get; set; }
+
+            public string Location
+            {
+                get { return _cmdlet.Location; }
+                set { _cmdlet.Location = value; }
+            }
+
+            string _defaultLocation = null;
+
+            public string DefaultLocation
+            {
+                get
+                {
+                    if (_defaultLocation == null)
+                    {
+                        var vmResourceType = _resourceClient.Providers.GetAsync("Microsoft.Compute").ConfigureAwait(false).GetAwaiter().GetResult()
+                        .ResourceTypes.Where(a => String.Equals(a.ResourceType, "virtualMachines", StringComparison.OrdinalIgnoreCase))
+                                      .FirstOrDefault();
+                        if (vmResourceType != null)
+                        {
+                            var availableLocations = vmResourceType.Locations.Select(a => a.ToLower().Replace(" ", ""));
```

Comment:
@markcowl do we also want to replace `-` characters? I doubt that ARM will change the format that they're returning locations in this case, but it might not hurt to canonicalize this string to be safe.